lib/module.js

```diff
@@ -41,9 +41,16 @@ module.exports = function (moduleOptions) {
       onlyFiles: true
     })
 
+    const toPascal = str => {
+      const camelCase = str.replace(/([-_]\w)/g, g => g[1].toUpperCase());
+      return camelCase[0].toUpperCase() + camelCase.substr(1)
+    }
+
     const globalComponents = fileNames.map(({ name, path }) => {
+      const origName = name.replace(new RegExp(`\\.(${suffixesWithExtensions.join('|')})$`), '');
```

Comment:
Not related to this PR but do we need to create a new `RegExp` on each iteration? Can we also move it to a cost in line above?